From 463f7cc1486b1154827ce0a8b519d4fb4ccd3253 Mon Sep 17 00:00:00 2001
From: Michel Pollet <buserror@gmail.com>
Date: Wed, 28 Nov 2012 16:04:39 +0000
Subject: [PATCH 05/11] imx23: Add 16 and 18 bits dtsi pin mapping

This specifies blocks for 16 and 18 bits
LCD pin mapping for the (LQFP) imx233

Signed-off-by: Claude Schwarz <claude.schwarz@gmail.com>
Signed-off-by: Michel Pollet <buserror@gmail.com>
---
 arch/arm/boot/dts/imx23.dtsi |   60 ++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 60 insertions(+)

diff --git a/arch/arm/boot/dts/imx23.dtsi b/arch/arm/boot/dts/imx23.dtsi
index 6d31aa3..67e2574 100644
--- a/arch/arm/boot/dts/imx23.dtsi
+++ b/arch/arm/boot/dts/imx23.dtsi
@@ -279,6 +279,66 @@
 					fsl,voltage = <1>;
 					fsl,pull-up = <0>;
 				};
+				
+				lcdif_16bit_pins_a: lcdif-16bit@0 {
+                                        reg = <0>;
+                                        fsl,pinmux-ids = <
+                                                0x1000 /* MX23_PAD_LCD_D00__LCD_D0 */
+                                                0x1010 /* MX23_PAD_LCD_D01__LCD_D1 */
+                                                0x1020 /* MX23_PAD_LCD_D02__LCD_D2 */
+                                                0x1030 /* MX23_PAD_LCD_D03__LCD_D3 */
+                                                0x1040 /* MX23_PAD_LCD_D04__LCD_D4 */
+                                                0x1050 /* MX23_PAD_LCD_D05__LCD_D5 */
+                                                0x1060 /* MX23_PAD_LCD_D06__LCD_D6 */
+                                                0x1070 /* MX23_PAD_LCD_D07__LCD_D7 */
+                                                0x0001 /* MX23_PAD_LCD_D08__LCD_D8 */
+                                                0x0011 /* MX23_PAD_LCD_D09__LCD_D9 */
+                                                0x0021 /* MX23_PAD_LCD_D10__LCD_D10 */
+                                                0x0031 /* MX23_PAD_LCD_D11__LCD_D11 */
+                                                0x0041 /* MX23_PAD_LCD_D12__LCD_D12 */
+                                                0x0051 /* MX23_PAD_LCD_D13__LCD_D13 */
+                                                0x0061 /* MX23_PAD_LCD_D14__LCD_D14 */
+                                                0x0071 /* MX23_PAD_LCD_D15__LCD_D15 */
+                                                0x1160 /* MX23_PAD_LCD_DOTCK__LCD_DOTCK */
+                                                0x1170 /* MX23_PAD_LCD_ENABLE__LCD_ENABLE */
+                                                0x1180 /* MX23_PAD_LCD_HSYNC__LCD_HSYNC */
+                                                0x1190 /* MX23_PAD_LCD_VSYNC__LCD_VSYNC */
+                                        >;
+                                        fsl,drive-strength = <0>;
+                                        fsl,voltage = <1>;
+                                        fsl,pull-up = <0>;
+                                };
+
+				lcdif_18bit_pins_a: lcdif-18bit@0 {
+                                        reg = <0>;
+                                        fsl,pinmux-ids = <
+                                                0x1000 /* MX23_PAD_LCD_D00__LCD_D0 */
+                                                0x1010 /* MX23_PAD_LCD_D01__LCD_D1 */
+                                                0x1020 /* MX23_PAD_LCD_D02__LCD_D2 */
+                                                0x1030 /* MX23_PAD_LCD_D03__LCD_D3 */
+                                                0x1040 /* MX23_PAD_LCD_D04__LCD_D4 */
+                                                0x1050 /* MX23_PAD_LCD_D05__LCD_D5 */
+                                                0x1060 /* MX23_PAD_LCD_D06__LCD_D6 */
+                                                0x1070 /* MX23_PAD_LCD_D07__LCD_D7 */
+                                                0x0001 /* MX23_PAD_LCD_D08__LCD_D8 */
+                                                0x0011 /* MX23_PAD_LCD_D09__LCD_D9 */
+                                                0x0021 /* MX23_PAD_LCD_D10__LCD_D10 */
+                                                0x0031 /* MX23_PAD_LCD_D11__LCD_D11 */
+                                                0x0041 /* MX23_PAD_LCD_D12__LCD_D12 */
+                                                0x0051 /* MX23_PAD_LCD_D13__LCD_D13 */
+                                                0x0061 /* MX23_PAD_LCD_D14__LCD_D14 */
+                                                0x0071 /* MX23_PAD_LCD_D15__LCD_D15 */
+                                                0x0101 /* MX23_PAD_LCD_D16__LCD_D16 */
+                                                0x0111 /* MX23_PAD_LCD_D17__LCD_D17 */
+                                                0x1160 /* MX23_PAD_LCD_DOTCK__LCD_DOTCK */
+                                                0x1170 /* MX23_PAD_LCD_ENABLE__LCD_ENABLE */
+                                                0x1180 /* MX23_PAD_LCD_HSYNC__LCD_HSYNC */
+                                                0x1190 /* MX23_PAD_LCD_VSYNC__LCD_VSYNC */
+                                        >;
+                                        fsl,drive-strength = <0>;
+                                        fsl,voltage = <1>;
+                                        fsl,pull-up = <0>;
+                                };
 			};
 
 			digctl@8001c000 {
-- 
1.7.10.4

